assets/src/tinymce/containers/Modal.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { withSelect, select } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import name from '../store/name';
+import WebStoriesModal from '../components/Modal';
+import { prepareShortCode } from '../utils';
+
+/**
+ *
+ * Pass extended props to the Modal component.
```

Comment:
Curious, what's the reason for doing the `withSelect`/`compose`/`mapSelectToProps` here?

Would it be easier to just use `useSelect` in `Modal`?

Reply:
Although, I haven't used `useSelect` extensively, but I think it can also be used without any problem.

`withSelect`  helps us keep data related operations separate from components, so that components stay in its purest form and also data agnostic. If I'm not wrong, usage of `useSelect` would need operations on store (calling store name and its method) which will somehow bind component to store.

This is only conceptual as we have couple of places in component where we are dispatching actions.

Also, `withSelect` and `withDispatch` are still in core and there seems no plan of deprecation for these APIs, I think we are good to go with these APIs.